indico_toolkit/indico_wrapper/dataset.py

```diff
@@ -11,44 +14,76 @@
-        self.dataset_id = dataset_id
 
-    @dataset_id_required
-    def get_dataset(self):
-        return self.client.call(GetDataset(self.dataset_id))
+    def get_dataset(self, dataset_id: int):
+        return self.client.call(GetDataset(dataset_id))
 
-    @dataset_id_required
-    def download_export(self, **kwargs) -> pd.DataFrame:
-        export_id = self._create_export(**kwargs)
+    def download_export(self, dataset_id: int, **kwargs) -> pd.DataFrame:
+        export_id = self._create_export(dataset_id, **kwargs)
         return self.client.call(DownloadExport(export_id))
 
-    def add_files_to_dataset(self, filepaths: List[str]) -> Dataset:
-        dataset = self._upload_files(filepaths)
-        return self._process_uploaded_files(dataset)
+    def add_files_to_dataset(self, dataset_id: int, filepaths: List[str]) -> Dataset:
+        """
+        Upload documents to an existing dataset and wait for them to OCR
+        """
+        dataset = self.client.call(
+            AddFiles(dataset_id=dataset_id, files=filepaths, wait=True)
+        )
+        datafile_ids = [f.id for f in dataset.files if f.status == "DOWNLOADED"]
+        return self.client.call(
+            ProcessFiles(dataset_id=dataset_id, datafile_ids=datafile_ids, wait=True)
+        )
+
+    def create_empty_dataset(
+        self, dataset_name: str, dataset_type: str = "DOCUMENT"
+    ) -> Dataset:
+        """
+        Create an empty dataset
+        Args:
+            name (str): Name of the dataset
+            dataset_type (str, optional): TEXT, IMAGE, or DOCUMENT. Defaults to "DOCUMENT".
+        """
+        return self.client.call(CreateEmptyDataset(dataset_name, dataset_type))
+
+    def create_large_doc_dataset(
+        self,
+        dataset_name: str,
+        filepaths: List[str],
+        batch_size: int = 3,
+        post_batch_delay_secs: int = 1,
+        verbose: bool = True,
+    ) -> Dataset:
+        """[summary]
+
+        Args:
+            dataset_name (str): create a name for the dataset 
+            filepaths (List[str]): files you want to upload
+            batch_size (int, optional): number of files to process at a time. Defaults to 3.
+            post_batch_delay_secs (int, optional): number of seconds to pause between batches. Defaults to 1.
+            verbose (bool, optional): print updates as each batch processes. Defaults to True.
+        """
+        fp = FileProcessing(filepaths)
+        dataset_id = self.create_empty_dataset(dataset_name).id
+        number_of_batches = int(math.ceil(len(filepaths) / batch_size))
+        for batch_fpaths in fp.batch_files(batch_size):
+            dataset = self.add_files_to_dataset(dataset_id, batch_fpaths)
+            if verbose:
+                number_of_batches -= 1
+                print(
+                    f"Finished batch at {datetime.now().time()}: {number_of_batches} remaining"
+                )
+            time.sleep(post_batch_delay_secs)
```

Comment:
What's the reason for the pause between batches? If the `add_files_to_dataset()` function is blocking until those files are completed I don't see the point.

Reply:
rationale was to allow more times for other peoples jobs to enter the queue, but I agree it probably doesn't make sense and will remove it